### ISSUE TITLE
clear text password in log file when ldap not available

### ISSUE THREAD

**Issue by dtkloc** [nextcloud/server#8104]
we had our ldap servers go offline and when users attempted to login thier password was captured in the log file

### Steps to reproduce
1.  configure ldap
2.  turn off ldap server
3. attempt to login

### Expected behaviour
it should log that you attempted to login but not list sensative information in logs

### Actual behaviour
password listed in clear text in the log file

### Server configuration

CentOs 7.3

Nginix
Postgres

**PHP version: 7.1

**Nextcloud version:** 12.0.5

**List of activated apps:**
collobora and ldap

**Comment by juliushaertl** [comment]
@nextcloud/ldap

**Comment by nickvergessen** [comment]
@dtkloc can you post the log entry with your cleared password, so we know where the problem is?

**Comment by blizzz** [comment]
@nickvergessen I can reproduce it,  this origins from the stack trace of ``OC\ServerNotAvailableException``, 

```
…
#14 [internal function]: OCA\User_LDAP\User_LDAP->checkPassword('user_1', 'user_1pwd')
…
#18 /srv/http/nextcloud/master/lib/private/User/Manager.php(204): OCA\User_LDAP\User_Proxy->checkPassword('user_1', 'user_1pwd')
#19 /srv/http/nextcloud/master/core/Controller/LoginController.php(262): OC\User\Manager->checkPasswordNoLogging('user_1', 'user_1pwd')
#20 [internal function]: OC\Core\Controller\LoginController->tryLogin('user_1', 'user_1pwd', NULL, false, 'Europe/Berlin', '1')
…
```

**Comment by dtkloc** [comment]
{"reqId":"4RXIpM8IMyG7nNE6dpnB","level":3,"time":"2017-09-19T13:29:05+00:00","remoteAddr":"IP_ADDRESS","user":"--","app":"index","method":"POST","url":"\/login","message":"Exception: {\"Exception\":\"OC\\\\ServerNotAvailableException\",\"Message\":\"Lost connection to LDAP server.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/LDAP.php(368): OCA\\\\User_LDAP\\\\LDAP->processLDAPError(Resource id #565)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/LDAP.php(295): OCA\\\\User_LDAP\\\\LDAP->postFunctionCall()\\n#2 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/LDAP.php(46): OCA\\\\User_LDAP\\\\LDAP->invokeLDAPMethod('bind', Resource id #565, '####PARTIAL BIND ACCOUNT NAME#####', '#####PARTIAL BIND PASSWORD####')\\n#3 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Connection.php(614): OCA\\\\User_LDAP\\\\LDAP->bind(Resource id #565, '####PARTIAL BIND ACCOUNT NAME#####', '#####PARTIAL BIND PASSWORD####')\\n#4 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Connection.php(539): OCA\\\\User_LDAP\\\\Connection->bind()\\n#5 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Connection.php(172): OCA\\\\User_LDAP\\\\Connection->establishConnection()\\n#6 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Connection.php(180): OCA\\\\User_LDAP\\\\Connection->init()\\n#7 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Access.php(1005): OCA\\\\User_LDAP\\\\Connection->getConnectionResource()\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Access.php(1166): OCA\\\\User_LDAP\\\\Access->executeSearch('(&(&(|(objectcl...', Array, Array, NULL, NULL)\\n#9 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Access.php(890): OCA\\\\User_LDAP\\\\Access->search('(&(&(|(objectcl...', Array, Array, NULL, NULL)\\n#10 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Access.php(806): OCA\\\\User_LDAP\\\\Access->searchUsers('(&(&(|(objectcl...', Array, NULL, NULL)\\n#11 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Access.php(780): OCA\\\\User_LDAP\\\\Access->fetchListOfUsers('(&(&(|(objectcl...', Array)\\n#12 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/User_LDAP.php(136): OCA\\\\User_LDAP\\\\Access->fetchUsersByLoginName('#####USER_ID#####', Array)\\n#13 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/User_LDAP.php(153): OCA\\\\User_LDAP\\\\User_LDAP->getLDAPUserByLoginName('####USER_ID####')\\n#14 [internal function]: OCA\\\\User_LDAP\\\\User_LDAP->checkPassword(*** sensitive parameters replaced ***)\\n#15 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/User_Proxy.php(71): call_user_func_array(Array, Array)\\n#16 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/Proxy.php(150): OCA\\\\User_LDAP\\\\User_Proxy->walkBackends('####USER_ID####', 'checkPassword', Array)\\n#17 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/User_Proxy.php(186): OCA\\\\User_LDAP\\\\Proxy->handleRequest('####USER_ID####', 'checkPassword', Array)\\n#18 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Manager.php(215): OCA\\\\User_LDAP\\\\User_Proxy->checkPassword(*** sensitive parameters replaced ***)\\n#19 \\\/var\\\/www\\\/nextcloud\\\/core\\\/Controller\\\/LoginController.php(231): OC\\\\User\\\\Manager->checkPasswordNoLogging('####USER_ID####', '####USER_PASSWORD####')\\n#20 [internal function]: OC\\\\Core\\\\Controller\\\\LoginController->tryLogin(*** sensitive parameters replaced ***)\\n#21 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)\\n#22 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#23 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(114): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#24 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OC\\\\\\\\Core\\\\\\\\Control...', 'tryLogin', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#25 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#26 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#27 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/base.php(1004): OC\\\\Route\\\\Router->match('\\\/login')\\n#28 \\\/var\\\/www\\\/nextcloud\\\/index.php(48): OC::handleRequest()\\n#29 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/user_ldap\\\/lib\\\/LDAP.php\",\"Line\":333}","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:55.0) Gecko\/20100101 Firefox\/55.0","version":"12.0.2.0"}

**Comment by blizzz** [comment]
Now with current master I cannot reproduce it anymore. Values are replaced as they should. Could you retry @dtkloc ?

**Comment by dtkloc** [comment]
I have verified that the issue is resolved. ThanksDavid
    On Thursday, March 15, 2018, 11:03:51 AM EDT, blizzz <notifications@github.com> wrote:  
 
 
Now with current master I cannot reproduce it anymore. Values are replaced as they should. Could you retry @dtkloc ?

**Comment by MorrisJobke** [comment]
LDAP stuff was fixed in 13.0.0: https://github.com/nextcloud/server/commit/b68609d0cf248ed42c9fe14706131da06b51d496 - see #6329

Let's close it then.
